Switch testing framework to fiveam

kilianmh opened the pull request:
- update `str.test.asd` to work with `fiveam`
- value to check against in first place, evaluated item in second place -> enables `fiveam` to output helpful error messages
- replace deprecated alias functions
- group functions in `suites`

How you like the names of the test-suites? And do you want to have some tests in other suites?

closes #94

kilianmh committed:
Switch testing framework to fiveam

- update str.test.asd to work with fiveam
- value to check against in first place, evaluated item in second place
  -> enables fiveam to output helpful error messages
- replace deprecated alias functions
- group functions in suites

vindarel commented:
Awesome, thank you so much.

You forgot the dev and test instructions in the README. (don't hesitate to show and explain fiveam commands: we can test with asdf:test-system, and also fivame:run!, and `(setf fiveam:*run-test-when-defined* t)` and then a C-c C-c on the test definition, and maybe more things that you used)

kilianmh committed:
Docs: update testing section

Update README testing section with information how to test the system.
This includes the main test suite, single test-suites, single tests, and
test after compilation.

kilianmh commented:
Now the _dev and test_ section is updated. Anything you do not like here?

One more thing: Should all test symbol (e.g. downcase) be exported or leave it like it is so that only the suite-symbols are exported?

vindarel commented:
No hard thoughts, I find it very fine like this. Thanks!
